Python 2/3 compatibility

boyska opened the pull request:
This pull request make the code compatible with both python2 and python3. To make a easy transition, I used the `future` library, which provides wrappers for python3 features into python2.

**NOTE**: I am _not_ proposing to officially support python3 from tomorrow, which would require some more consideration. But making a single codebase future-proof is an important goal, I believe.
### How to test
#### Python2

``` sh
mkvirtualenv -p /usr/bin/python2 fsdb-2
python setup.py 'test'
```
#### Python3

``` sh
mkvirtualenv -p /usr/bin/python3 fsdb-3
python setup.py 'test'
```

boyska committed:
Python 2/3 compatibility

ael-code commented on fsdb/Fsdb.py:
```diff
@@ -49,7 +56,7 @@ def __init__(self, fsdbRoot, mode=None, deep=None, hash_alg=None):
 
         # on different platforms same unicode string could have different rappresentation
-        if isinstance(fsdbRoot, unicode):
+        if isinstance(fsdbRoot, str):
```
`if isinstance(fsdbRoot, str):` this works only on python3

On [python2](https://docs.python.org/2/howto/unicode.html#python-2-x-s-unicode-support) str and unicode are different class

boyska replied:
yes, but on line 7 I do `from builtins import str` which will change the names to be the same as in python3.
Anyway, I think that the best solution is to add unit tests and see if this pull request will make them fail.

ael-code commented:
> in doing so, some refactoring has been necessary: unit test directory is now randomly generated. It was a problem because previously if I had a bug in the code, any subsequent unittest would have failed because of that. Also, it was using '/tmp', which is very platform-specific. I am now using the tempfile standard python module.

should be solved in 5689612cce6345ca1f34cc271784958985e144e1.

ael-code committed:
log everuy corrupted file found

ael-code committed:
added __getitem__() function and relative tests

ael-code committed:
updated docs

boyska commented:
I rebased the code upon your latest changes. I also removed the mention about the unittests thing, since it's not contained in this pull request anymore.

ael-code commented:
python 3 compatibility has been added since version `1.0`.
I've preferred to not use your contribution because you were proposing to add `future` as dependecy.
Thanks anyway
